src/quickfix.c

```diff
@@ -8044,6 +8045,9 @@ set_errorlist(
 	    qf_list_changed(qf_get_curlist(qi));
     }
 
+    qfbuf = buflist_findnr(qi->qf_bufnr);
+    apply_autocmds(EVENT_QUICKFIXCMDCHANGED, NULL, NULL, FALSE, qfbuf);
```

Comment:
What is the use case for this autocmd?  What happens if the quickfix list or location list is modified from this autocmd?

Reply:
Currently, you can use the `QuickFixCmdPost` event to, e.g., open a quickfix or location list window if a command has filled the list with something to work on. (For example, [here](https://gist.github.com/romainl/ce55ce6fdc1659c5fbc0f4224fd6ad29#automatic-opening-of-the-quickfix-window).) But if you fill the lists using `setqflist` or `setloclist`, then the `QuickFixCmdPost` event is not triggered. In that case, you have to manually create and fire a dummy autocommand for users to hook into. E.g., [here](https://github.com/neovim/neovim/issues/13099#issuecomment-2541490941). This additional autocommand event would help users and plugin authors that fill the quickfix or location lists with `setqflist` and `setloclist`.

Reply:
after use setqflist or setloclist before `copen lopen` in this case qfbuf is NULL do we want also trigger ? i do trigger here

Reply:
@telemachus  moved to trigger on qf_list_changed. But this function runs before qf_update_buffer. Meaning that when the event triggers the qfbuffer may not be updated. Does this have any impact.